### ISSUE TITLE
[Feature Suggestion] Allow disabling analytics rules and ignoring directories

### ISSUE THREAD

**Issue by emilgoldsmith** [elm-tooling/elm-language-server#606]
## Expected Behavior
Hi! Thank you so much to all the contributors for the work you do! I may be able to contribute to this change in the future, though not right now, but just wanted to log this feature suggestion now.

My specific needs stem from:
- Wanting to disable the single field record analytics rule as I think it can add a lot of important context to have a named argument in many circumstances, but I don't want to fully disable the analytics as then I for example lose the unused imports greying out in the editor etc.
- Wanting to ignore .elm-spa/generated which Elm SPA automatically generates so I don't have `problems` with missing type annotations showing up in my VSCode `Problems` tab 

## Current Behavior
Right now I can only enable or disable all analytics rules at once and only for all files in my repo

## Possible Solution
Think it's obvious from above :), for "related work" Elm Analytics has these features so maybe code / implementation strategies could be borrowed from there

**Comment by Razzeee** [comment]
You can use elm-analyse.json to disable rules and ignore directories . See
their documentation (yes, we also need to have that on our side)

That being said, the rule your speaking of, will default to off in the next
release.

Emil Goldsmith Olesen ***@***.***> schrieb am Do., 10. Juni
2021, 11:44:

**Comment by emilgoldsmith** [comment]
Great about the default for the rule, but I already have an
elm-analyse.json and it works fine when I run elm analyse for example in
CI, but those errors still show in vscode

**Comment by Razzeee** [comment]
Is the file next to the elm.json?

Emil Goldsmith Olesen ***@***.***> schrieb am Do., 10. Juni
2021, 15:14:

> Great about the default for the rule, but I already have an
> elm-analyse.json and it works fine when I run elm analyse for example in
> CI, but those errors still show in vscode
>

**Comment by emilgoldsmith** [comment]
Yep! At the root

**Comment by Razzeee** [comment]
Can you setup an example repo where this reproduces? It's really hard to see why this fails otherwise.

**Comment by emilgoldsmith** [comment]
I’ll try making the time, and get back to you if I do!

**Comment by hanshenrik** [comment]
I'm also interested in this. We have an `elm-analyse.json` file that looks like this:

```
{
    "checks": {
        "ExposeAll": false,
        "ImportAll": false,
        "SingleFieldRecord": false,
        "TriggerWords": false,
        "UseConsOverConcat": false,
        "UnnecessaryParens": false
    },
    "excludedPaths": [
        "generated_api_src"
    ]
}
```

The plugin works fine up until version 1.5.3, but for versions 2.0.0 and above I get a bunch of errors listed in the `Problems` tab in VSCode. I guess this is because it no longer ignores the folders listed in `excludedPaths`. Do you see anything wrong with the `elm-analyse.json` that should cause this?

**Comment by Razzeee** [comment]
No, in fact I just setup a test repo with that config and was able to confirm, that excludedPaths work on Linux with the latest release.

**Comment by hanshenrik** [comment]
Thanks for the quick response. I figured out what my problem was (not really related to this issue, but perhaps useful nonetheless): Turned out I had an old `elm-format` version install globally (v0.8.2). We also have a version installed locally in the project (v0.8.5). Since we have a monorepo, the binary is not located at `node_modules/elm-format/bin`, but rather at `frontend/node_modules/elm-format/bin`. I guess this was the issue, since the "Elm LS: Elm Format Path" config option says it should look for local versions first if its left empty. So I had to change this to `frontend/node_modules/elm-format/bin`, and now the errors are gone :)
